src/search/engines/bing/bing.go

```diff
@@ -70,28 +60,40 @@ func Search(ctx context.Context, query string, relay *bucket.Relay, options engi
 		}
 	})
 
-	localeParam := getLocale(&options)
+	localeParam := getLocale(options)
+
+	retErrors := make([]error, options.MaxPages)
 
 	colCtx := colly.NewContext()
 	colCtx.Put("page", strconv.Itoa(1))
 
 	urll := Info.URL + query + localeParam
 	anonUrll := Info.URL + anonymize.String(query) + localeParam
-	_sedefaults.DoGetRequest(urll, anonUrll, colCtx, col, Info.Name, &retError)
+
+	err = _sedefaults.DoGetRequest(urll, anonUrll, colCtx, col, Info.Name)
+	retErrors[0] = err
 
 	for i := 1; i < options.MaxPages; i++ {
 		colCtx = colly.NewContext()
 		colCtx.Put("page", strconv.Itoa(i+1))
 
 		urll := Info.URL + query + "&first=" + strconv.Itoa(i*10+1) + localeParam
 		anonUrll := Info.URL + anonymize.String(query) + "&first=" + strconv.Itoa(i*10+1) + localeParam
-		_sedefaults.DoGetRequest(urll, anonUrll, colCtx, col, Info.Name, &retError)
+
+		err = _sedefaults.DoGetRequest(urll, anonUrll, colCtx, col, Info.Name)
+		retErrors[i] = err
 	}
 
 	col.Wait()
 	pagesCol.Wait()
 
-	return retError
+	realRetErrors := make([]error, 0)
+	for _, err := range retErrors {
+		if err != nil {
+			realRetErrors = append(realRetErrors, err)
+		}
+	}
+	return realRetErrors
```

Comment:
Please pull this out to a function (its used everywhere). 
Not sure if this is the best way to do this, it probably allocates extraneous heap memory, though it might be the most readable solution (as opposed to only making one slice and only appending non-zero errors to it, potentially with a helper `appendNonNil` function or something) - its up to you.
Also it might be better if the slice is given some non-zero capacity (and zero length).

Reply:
In 99% of cases it doesn't allocate ANY memory since the len and cap of the realRetErrors will stay 0, since we expect that (currently) all errors returned are nil (since the errors are only from invoking the colly Async request, and the actuall errors happening are handled by col.Error).